octodns_googlecloud/__init__.py

```diff
@@ -69,20 +69,25 @@ def _apply(self, plan):
 
             if class_name == 'Create':
                 gcloud_changes.add_record_set(
-                    _rrset_func(gcloud_zone, change.record))
+                _rrset_func(gcloud_zone, change.record))
             elif class_name == 'Delete':
                 gcloud_changes.delete_record_set(
-                    _rrset_func(gcloud_zone, change.record))
+                _rrset_func(gcloud_zone, change.record))
             elif class_name == 'Update':
                 gcloud_changes.delete_record_set(
-                    _rrset_func(gcloud_zone, change.existing))
+                _rrset_func(gcloud_zone, change.existing))
                 gcloud_changes.add_record_set(
-                    _rrset_func(gcloud_zone, change.new))
+                _rrset_func(gcloud_zone, change.new))
             else:
                 msg = f'Change type "{class_name}" for change ' \
-                    f'"{str(change)}" is none of "Create", "Delete" or "Update'
+                    f'"{str(changes)}" is none of "Create", "Delete" or "Update'
                 raise RuntimeError(msg)
 
+            if len(gcloud_changes.additions) == 1000 or len(gcloud_changes.deletions) == 1000:
+                self._really_apply(self, gcloud_changes)
+                gcloud_changes.reload()
```

Comment:
This looks like this submits things when there is 1000 changes. What triggers the submissions (`create`) when there's fewer than 1k? E.g. https://github.com/octodns/octodns-route53/blob/e5e529e0bb46be1cf2690f98faaa4084bf6dc109/octodns_route53/__init__.py#L1543-L1550 for `Route53Provider`

Reply:
I just pushed an updated version with a _first_ approach to batches. So, splitting additions, deletions and updates in batches. This is based in sorting changes into a dict. Once again, it's a first approach. Is this kind of review and update fashion ok for you? My coding might not be world-class but it's a good exercise for me.

Reply:
You'll notice that the Changes object will only contain either additions, deletions or updates (add + del). I feel that trying to combine everything together might be overkill but maybe I'm just not seeing a good way to efficiently putting everything together.